graphql-ballerina/src/graphql/parser/lexer.bal

```diff
@@ -0,0 +1,325 @@
+            self.inProgress = false;
+            return getTokenFromChar(char);
+        } else if (tokenType == T_STRING) {
+            return self.readStringToken(char.location);
+        } else if (tokenType == T_INT) {
+            return self.readNumeralToken(char.location, char.value);
+        } else if (tokenType is TerminalCharacter) {
+            return getTokenFromChar(char);
+        } else if (tokenType == T_COMMENT) {
+            return self.readCommentToken(char.location);
+        } else if (tokenType is SpecialCharacter) {
+            return getTokenFromChar(char);
+        } else {
+            return self.readWordToken(char);
+        }
+    }
+
+    isolated function getNextSpecialCharaterToken() returns Token|ParsingError {
+        Token token = check self.next();
+        if (token.'type is SpecialCharacter) {
+            return self.getNextSpecialCharaterToken();
+        } else {
+            return token;
+        }
+    }
+
+    isolated function readStringToken(Location location) returns Token|SyntaxError {
+        string previousChar = "";
+        string word = "";
+        while (!self.charReader.isEof()) {
+            Char charToken = self.charReader.next();
+            string value = charToken.value;
+            if (value is EOF) {
+                return getUnexpectedTokenError(getTokenFromChar(charToken));
+            }
+            if (value is LineTerminator) {
+                string message = "Syntax Error: Unterminated string.";
+                ErrorRecord errorRecord = {
+                    locations: [location.clone()]
+                };
+                return UnterminatedStringError(message, errorRecord = errorRecord);
+            } else if (value is QUOTE && previousChar != BACK_SLASH) {
+                break;
+            } else {
+                word += value;
+            }
+            previousChar = value;
+        }
+        return getToken(word, T_STRING, location);
+    }
+
+    isolated function readNumeralToken(Location location, string fisrtChar) returns Token|ParsingError {
+        string numeral = fisrtChar;
+        boolean isFloat = false;
+        while (!self.charReader.isEof()) {
+            Char char = self.charReader.peek();
+            TokenType tokenType = getTokenType(char);
+            string value = char.value;
+            if (tokenType is TerminalCharacter) {
+                break;
+            } else if (tokenType is SpecialCharacter) {
+                break;
+            } else if (value == DECIMAL) {
+                char = self.charReader.next();
+                numeral += value;
+                isFloat = true;
+            } else if (value is Numeral) {
+                char = self.charReader.next();
+                numeral += value.toString();
+            } else {
+                char = self.charReader.next();
+                string message = "Syntax Error: Invalid number, expected digit but got: \"" + value + "\".";
```

Comment:
Saying "Syntax Error" may not be that meaningful, since this is more of a payload parsing, rather than a language syntax parsing.

Reply:
This is the standard error message format in GraphQL implementation in ExpressJS.

Reply:
We can follow our own format, since we have a full control over the impl.

Reply:
But the issue is this is the convention followed by original GraphQL implementors, and also the GO GraphQL library. I think it's better to keep up with that, since otherwise, it may confuse the users familiar with these errors. WDYT?

Reply:
Yea, its good to stick to the conventions, but if we think from ballerina point of view, a graphQL request is very similar to a JSON/XML request and the parsing also should be similar to parsing an XML or a JSON. And in those two scenarios we don't really call an error occurred during payload parsing as a syntax error, do we? IMO we should give a unified ballerina-ish error for all sorts of payload parsing errors.
The term 'syntax error' has it own meaning, hence I would prefer we don't mix it with runtime errors.